pkg/cloudevents/transport/amqp/transport_test.go

```diff
@@ -0,0 +1,108 @@
+package amqp_test
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/pkg/binding/spec"
+	"github.com/cloudevents/sdk-go/pkg/binding/test"
+	ce "github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport/amqp"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/stretchr/testify/assert"
+)
+
+// TODO(alanconway) Need better self-test without external dependency.
+
+// Requires an external AMQP broker or router, skip if not available.
+// The env variable TEST_AMQP_URL provides the URL, default is "/test"
+//
+// One option is http://qpid.apache.org/components/dispatch-router/indext.html.
+// It can be installed from source or from RPMs, see https://qpid.apache.org/packages.html
```

Comment:
did that, locally: worked fine.

@alanconway can you capture that in a future issue, that perhaps Circle CI can spin up a (docker based) Dispatch Routcer?

Reply:
https://github.com/cloudevents/sdk-go/issues/236